notion-core/src/test/scala/com/ambiata/notion/core/LocationTemporary.scala

```diff
@@ -43,6 +43,16 @@ case class LocationTemporary(t: T, path: String, client: AmazonS3Client, conf: C
 }
 
 object LocationTemporary {
+
+  /** @return an arbitrary location where the path possibly ends with a / */
   implicit def LocationTemporaryArbitrary: Arbitrary[LocationTemporary] =
-    Arbitrary(arbitrary[T].map(LocationTemporary(_, java.util.UUID.randomUUID().toString, Clients.s3, new Configuration)))
+    Arbitrary {
+      val path = java.util.UUID.randomUUID.toString
+
+      arbitrary[T].flatMap { t =>
+        val locationTemporary = LocationTemporary(t, path, Clients.s3, new Configuration)
+        val locationDirTemporary = LocationTemporary(t, path + "/", Clients.s3, new Configuration)
```

Comment:
As discussed with @nhibberd in person, a more detailed example of my "breakage pattern"

Two keys with content
```
foo.txt
foo.txt/nested.text
```

This before would of returned `true` for is directory given the input `foo.txt`, but with this change would return `false`. This is more problematic at the moment because this exact pattern is being generated by memento placement in an internal project.